### PR TITLE
Feature/13 show doctors

### PR DESCRIPTION
**Pull request by ankestein**
Get all doctors:
Add tests and add table to frontend

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/service/DoctorService.java [hunk 1]
```diff
@@ -50,4 +55,47 @@ public Doctor addDoctor(DoctorDto doctorDto) {
         }
     }
 
+
+    public Doctor addAppointment(DoctorDto doctorDto) {
```
Comment (macben888) [inline backend/src/main/java/de/neuefische/backend/service/DoctorService.java:59]: this method is way to long try and refactor it into smaller portions that'll make it easier to read/understand

File: frontend/src/service/DoctorApiService.js [hunk 2]
```diff
@@ -3,3 +3,13 @@ import axios from 'axios';
 export function submitDoctor(newDoctor) {
 	return axios.post('/api/doctor', newDoctor).then((response) => response.data);
 }
+
+export function submitAppointment(newDoctor) {
+	return axios
+		.put('/api/doctor/appointment', newDoctor)
```
Comment (macben888) [inline frontend/src/service/DoctorApiService.js:9]: technically not rest conform to my understanding. you're updating appointment attribute of the doctor, so it should be put(api/doctor/doctorId, newDoctor)
Maybe not so important after all...

**Review by macben888 — APPROVED** [review]

File: backend/src/test/java/de/neuefische/backend/controller/DoctorControllerTest.java [hunk 3]
```diff
@@ -39,30 +51,98 @@ void addDoctorTest() {
...
+
+        // WHEN
+        ResponseEntity<Doctor> response = testRestTemplate.postForEntity("/api/doctor", doctorDto, Doctor.class);
+        Doctor actual = response.getBody();
+        String actualId = actual.getId();
+
+        // THEN
+        expectedDoctor.setId(actualId);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(expectedDoctor, actual);
```
Comment (Flooooooooooorian) [inline backend/src/test/java/de/neuefische/backend/controller/DoctorControllerTest.java:64]:
```suggestion
        ResponseEntity<Doctor> response = testRestTemplate.postForEntity("/api/doctor", doctorDto, Doctor.class);
       
        // THEN
        assertEquals(HttpStatus.OK, response.getStatusCode());
        //assert(id, notNull)
        
        
        Doctor actual = response.getBody();
        expectedDoctor.setId(actual.getId());

        
        assertEquals(expectedDoctor, actual);
```

File: backend/src/test/java/de/neuefische/backend/service/DoctorServiceTest.java [hunk 4]
```diff
@@ -80,4 +107,230 @@ void addExistingDoctorTest() {
...
+        when(doctorRepo.save(expectedDoctor)).thenReturn(expectedDoctor);
+        when(utilService.mapDoctorDtoToDoctor(expectedDoctorDto)).thenReturn(expectedDoctor);
+        when(utilService.mapAppointmentDtoToAppointment(expectedDoctorDto.getAppointmentDto())).thenReturn(expectedDoctor.getAppointments().get(0));
+
+        // WHEN
+        Doctor actualDoctor = doctorService.addAppointment(expectedDoctorDto);
+
+        // THEN
+        assertEquals(expectedDoctor, actualDoctor);
+        verify(doctorRepo).save(expectedDoctor);
```
Comment (Flooooooooooorian) [inline backend/src/test/java/de/neuefische/backend/service/DoctorServiceTest.java:149]: verify all mocks
```suggestion
        verify(doctorRepo).save(expectedDoctor);
```

File: backend/src/test/java/de/neuefische/backend/service/DoctorServiceTest.java [hunk 5]
```diff
@@ -80,4 +107,230 @@ void addExistingDoctorTest() {
...
+        when(idService.generateId()).thenReturn("123456");
+        when(doctorRepo.findDoctorByFirstNameAndLastNameAndSpecialtyAndCity("Linda", "Holder", "Dentist", "Bonn")).thenReturn(doctorInDb);
+        when(doctorRepo.save(doctorInDb)).thenReturn(updatedDoctor);
+
+        // WHEN
+        Doctor actualDoctor = doctorService.addAppointment(doctorDto);
+
+        // THEN
+        assertEquals(updatedDoctor, actualDoctor);
+        verify(doctorRepo).save(doctorInDb);
```
Comment (Flooooooooooorian) [inline backend/src/test/java/de/neuefische/backend/service/DoctorServiceTest.java:268]: verify all mock calls
```suggestion
        verify(doctorRepo).save(doctorInDb);
```

File: frontend/src/pages/Doctors.js [hunk 6]
```diff
@@ -1,21 +1,67 @@
 import AddIcon from '@mui/icons-material/Add';
 import {Fab} from '@mui/material';
+import {DataGrid} from '@mui/x-data-grid';
 import {Link} from 'react-router-dom';
+import {useEffect, useState} from 'react';
+import {getDoctors} from '../service/DoctorApiService';
+import styled from 'styled-components/macro';
 
 export default function Doctors() {
+	const [doctors, setDoctors] = useState([]);
```
Comment (Flooooooooooorian) [inline frontend/src/pages/Doctors.js:10]: Move to App.js

**Review by Flooooooooooorian — APPROVED** [review]

**Pull request merged**